client/src/main/java/org/opentestsystem/authoring/testspecbank/client/TestSpecBankClient.java

```diff
@@ -70,6 +100,45 @@ public TestSpecificationSearchResponse getTestSpecificationByTenantSet(final Set
         return response;
     }
 
+    @Override
+    public Optional<ValidationError> deleteTestSpecification(final String testSpecificationKey) {
+        Optional<ValidationError> maybeValidationError = Optional.empty();
+        final URI uri =
+            UriComponentsBuilder.fromUriString(String.format("%stestSpecification/%s", baseUri, testSpecificationKey))
+                .build()
+                .toUri();
+
+        try {
+            tsbOauthRestTemplate.delete(uri);
+        } catch (final HttpClientErrorException hce) {
+            LOGGER.error("Error deleting {0} test specification: ", hce);
+            if (hce.getStatusCode() == HttpStatus.UNPROCESSABLE_ENTITY) {
+                try {
```

Comment:
This nested `try` is a bit messy.  I had to wrap the `objectMapper` code in its own `try... catch` to handle the exception it might throw.  Otherwise I'd have to add `throws Exception` to the method signature.  Since I'm of the opinion that this client should never throw an exception, this was the only way I could "cleanly" do it.